common/djangoapps/student/models.py

```diff
@@ -1436,11 +1443,17 @@ def enrollment_mode_for_user(cls, user, course_id):
             and is_active is whether the enrollment is active.
         Returns (None, None) if the courseenrollment record does not exist.
         """
-        try:
-            record = cls.objects.get(user=user, course_id=course_id)
-            return (record.mode, record.is_active)
-        except cls.DoesNotExist:
-            return (None, None)
+        cached_value = cls._get_enrollment_in_request_cache(user, course_id)
+        if cached_value:
+            mode, is_active = cached_value
+        else:
+            try:
+                record = cls.objects.get(user=user, course_id=course_id)
+                mode, is_active = (record.mode, record.is_active)
+            except cls.DoesNotExist:
+                mode, is_active = (None, None)
+            cls._update_enrollment_in_request_cache(user, course_id, mode, is_active)
+        return mode, is_active
```

Comment:
Why split up the tuple just to return both parts here?  Could `cls._update_enrollment_in_request_cache` take the tuple as one argument instead of two?